server/api/cli/cli.controller.js

```diff
@@ -39,17 +39,22 @@ exports.run = function (req, res) {
   dirName += dateTime.getFullYear()
   const execFile = req.body.fileName.replace('.c', '')
   const cmd = `"${dirName}/${execFile}"`
-  exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
+  var cp = exec(cmd, { timeout: 10000 }, // Process will time out if running for > 10 seconds.
     (error, stdout, stderr) => {
       if (error) {
-        return res.send(200, error)
+        return res.send(200, {"info":error, "process":cp.pid});
```

Comment:
Yes I was confused about that also. No, the server comes back with an error right away. I was asking Celina about this (I had originally thought it would take 10 sec to respond) but she said that it comes back with an error, but that the process is still running in the background - so I focused on killing the process, but still not sure why the timeout isn't being activated (and an error is generated right away)